main.js

```diff
@@ -769,17 +893,56 @@ function updateIniFromForm(ini) {
     }
 }
 
-function updateSelection(id, value, list) {
+function updateSelection(id, value, newChoiceArray) {
     "use strict";
-    value = parseInt(value.trim());
-    while (document.getElementById(id).firstChild) {
-        document.getElementById(id).removeChild(document.getElementById(id).firstChild);
+    var select = document.getElementById(id);
+    
+    // Sanitize value
+    if (typeof value === "string") {
+        value = parseInt(value.trim());
+    } else if (typeof value === "number") {
+        value = parseInt(value);
+    } else if (value == undefined) {
+        if (select.value == undefined) {
+            window.alert("No value found for form input " + id + ", defaulting to 0.");
+            value = 0;
+        } else {
+            value = select.value;
+        }
+    }
+    
+    // Switch selected array
+    if (newChoiceArray) {
+        stateChoiceArrays[id] = newChoiceArray;
     }
-    if (!list[value]) {
-        list[value] = "Unrecognized (" + value + ")";
+    
+    // Default case if incorrect ID is used (or stateChoiceArrays missing a case)
+    if (!stateChoiceArrays[id]) {
+        window.alert("No associated array for form input " + id + " found, defaulting to [\"Error\"].");
+        stateChoiceArrays[id] = ["Error"];
+    }
+    
+    // Add "Unrecognized" value if necessary
+    if (!stateChoiceArrays[id][value]) {
+        stateChoiceArrays[id][value] = "Unrecognized (" + value + ")";
     }
-    loadSelectFromObj(id, list);
-    document.getElementById(id).value = value;
+    
+    // Clear old options
+    while (select.firstChild) {
+        select.removeChild(select.firstChild);
+    }
+    
+    // Create options
+    for (var key of Object.keys(stateChoiceArrays[id]).sort((a, b) => a - b)) { // (Decimal keys don't automatically sort correctly)
```

Comment:
Do you have Internet Explorer nearby to test whether Flowey's Time Machine works correctly on it? Because I would rather not drop support for it if it was already supported.

Reply:
It took a reboot, but now I do. And... it doesn't work. `display: grid` seems the obvious problem, but, more importantly, the scripts aren't running. I don't know IE's dev tools well enough to find a fix for that if an easy one exists.

Reply:
Okay, looks like the string literal at line 982 is the issue. Not sure if there are more, but combined with the total display failure I think we can safely say IE is no longer supported.